tests/test_busybox.py

```diff
@@ -48,6 +48,42 @@ def test_busybox_image_size(
     )
 
 
+@pytest.mark.parametrize(
+    "container_per_test", [BUSYBOX_CONTAINER], indirect=True
+)
+def test_busybox_links(container_per_test):
+    # Ensure all commands in /bin are links to /usr/bin/busybox
+    container_per_test.connection.run_expect(
+        [0],
+        'for i in /bin/*; do stat -c "%N" "$i" | grep "/usr/bin/busybox"; done',
+    )
+
+
+@pytest.mark.parametrize(
+    "container_per_test", [BUSYBOX_CONTAINER], indirect=True
+)
+def test_busybox(container_per_test):
+    # Test only a selection of all busybox tools
+    busybox = container_per_test.connection.run_expect([0], "busybox")
+    container_per_test.connection.run_expect([0], "true")
```

Comment:
I don't get these checks, but it might be due to my lack of experience with busybox.
How running `true` cmd is special for busybox and not for any random linux environment? `true` will always return 0...
Same for `echo 'test' | cat | grep 'test'` and other checks.. you can run this on any linux env and get the same result, why does it special for busybox?

Reply:
BusyBox is a multi-call binary, an executable program that performs the same job as more than one utility program. That means there is just a single BusyBox binary, but that single binary acts like a large number of utilities. Look at the DESCRIPTION section in https://busybox.net/downloads/BusyBox.html.

All of those program are just symlinks to `/usr/bin/busybox`, see e.g.

    # ls -al /bin
    total 328
    drwxr-xr-x    1 root     root           796 May  8 08:22 .
    drwxr-xr-x    1 root     root           136 May  9 08:18 ..
    lrwxrwxrwx    1 root     root            16 May  8 00:36 arch -> /usr/bin/busybox
    lrwxrwxrwx    1 root     root            16 May  8 00:36 arping -> /usr/bin/busybox
    lrwxrwxrwx    1 root     root            16 May  8 00:36 awk -> /usr/bin/busybox
    ...
    lrwxrwxrwx    1 root     root            16 May  8 00:36 cat -> /usr/bin/busybox
    ...
    lrwxrwxrwx    1 root     root            16 May  8 00:36 true -> /usr/bin/busybox
    ...

In the busybox image, when you run `cat` it does not execute the `/usr/bin/cat` program as standalone, but runs the `busybox` binary and the `cat` applet therein.

The test now runs a bunch of commands to ensure, they are integrated well in the single `busybox` binary by running a subset of those applets and check the expected output, if they are functional and correct.

Reply:
Yes, that I had some idea, looking at `test_busybox_links` test, however you are using here a small subset, and I was wondering if there is anything special in these commands you are using, and why not using other things like .e.g `ln` or `ls` ?

Reply:
No there is nothing special, those were just low hanging fruit to check if the integration is working properly.

The main point of this PR is to make sure that the system utilities are coming from the `busybox` binary and not from accidentally packaged programs. The rest is just a smoketest to ensure, that `busybox` is working properly.

Reply:
Ok, as agreed in the chat. In next iteration, we could have the list of expected binaries hardcoded here and make asserts for each one of them to check they exist in the container.